README.md

```diff
@@ -83,6 +83,41 @@ func (s *contactsServer) Read(ctx context.Context, req *ReadRequest) (*ReadRespo
 
 When bootstrapping a gRPC server, add middleware that will extract the account_id token from the request context and set it in the request struct. The middleware will have to navigate the request struct via reflection, in the case that the account_id field is nested within the request (like if it's in a request wrapper as per our example above)
 
+##### Transaction Management
+
+We provide transaction management by offering `gorm.Transaction` wrapper and `gorm.UnaryServerInterceptor`.
+The `gorm.Transaction` works as a singleton to prevent an application of creating more than one transaction instance per incoming request.
+The `gorm.UnaryServerInterceptor` performs management on transactions. 
+Interceptor creates new transaction on each incoming request and commits it if request finishes without error, otherwise transaction is aborted.
+The created transaction is stored in `context.Context` and passed to the request handler as usual.
+
+```go
+// add gorm interceptor to the chain
+  server := grpc.NewServer(
+    grpc.UnaryInterceptor(
+      grpc_middleware.ChainUnaryServer( // middleware chain
+        ...
+        gorm.UnaryServerInterceptor(), // transaction management
+        ...
+      ),
+    ),
+  )
+
+import (
+	"github.com/infobloxcopen/atlas-app-toolkit/gorm"
+)
+
+func (s *MyService) MyMethod(ctx context.Context, req *MyMethodRequest) (*MyMethodResponse, error) {
+	// extract gorm transaction from context
+	txn, ok := gorm.FromContext(ctx)
```

Comment:
(minor) Guess it would be nice to have more self-explaining names for these funcs.
E.g. ` transaction.NewContext( )` or `gorm.NewTransactionContext( )` instead of `gorm.NewContext( )`

Reply:
That is common naming scheme for context related functions.
Strictly speaking transaction is a `gorm.DB` instance and `gorm.NewContext()` with opposite `gorm.FromContext()`  looks convenient and self descriptive FMPOV.

I intentionally did not add separate `transaction` package because it requires definition of generic `Transaction` interface that performs transaction management and does not depend on any of underlying implementation (sql.Tx, gorm, etc..)